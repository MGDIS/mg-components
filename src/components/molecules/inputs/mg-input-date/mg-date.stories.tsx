import { h } from "@stencil/core";

export default {
  component: 'mg-input-date',
  title: 'Molecules/mg-input-date',
  parameters: {
    docs: {
      source: {
        code: `<mg-input-date label="Label" required has-colon></mg-input-date>`,
      },
    },
  },
};

/**
 * 1. camelCase arguments must be written in the template, for exemple labelOnTop must be placed in the template as label-on-top={args.labelOnTop}
 * 2. boolean arguments with a default true value must be added like display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
 */
const Template = args => (
  <mg-input-date
    {...args}
    label-on-top={args.labelOnTop}
    label-colon={args.labelColon}
    pattern-error-message={args.patternErrorMessage}
    display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
    character-left-template={args.characterLeftTemplate}
    help-text={args.helpText}
  ></mg-input-date>
);

export const MgInputDate = Template.bind({});
MgInputDate.args = {
  // Global
  value: '2021-10-14',
  identifier: 'reference-id',
  name: 'input-name',
  // Label
  label: 'Label',
  labelOnTop: false,
  labelColon: true,
  // Input
  required: true,
  readonly: false,
  disabled: false,
  // Tooltip
  tooltip: 'This is a tooltip',
  // Help Text
  helpText: 'Help text with html <strong>bold</strong>, <em>italic</em>.',
};
