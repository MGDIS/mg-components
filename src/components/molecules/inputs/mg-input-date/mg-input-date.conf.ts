export enum InputError {
  MIN = 'min',
  MAX = 'max',
  MINMAX = 'minMax',
  REQUIRED = 'required',
}
