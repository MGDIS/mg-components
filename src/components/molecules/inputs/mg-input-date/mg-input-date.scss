@use '../styles/mg-input.scss';
@use '../styles/mg-input-box.scss';
@use '../../../../styles/a11y.scss';
