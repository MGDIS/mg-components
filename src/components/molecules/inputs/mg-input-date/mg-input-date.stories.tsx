import { h } from '@stencil/core';

export default {
  component: 'mg-input-date',
  title: 'Molecules/Inputs/mg-input-date',
};

/**
 * 1. camelCase arguments must be written in the template, for exemple labelOnTop must be placed in the template as label-on-top={args.labelOnTop}
 * 2. boolean arguments with a default true value must be added like display-character-left={args.displayCharacterLeft ? 'true' : 'false'}
 */
const Template = args => {
  const labelOnTop = args.labelOnTop;
  delete args.labelOnTop;
  const labelHide = args.labelHide;
  delete args.labelHide;
  const helpText = args.helpText;
  delete args.helpText;
  // return element
  return <mg-input-date {...args} label-on-top={labelOnTop} label-hide={labelHide} help-text={helpText}></mg-input-date>;
};

export const MgInputDate = Template.bind({});

const date = new Date();
const getMonth = date => {
  const month = `${date.getMonth() + 1}`;
  return `${month.length > 1 ? month : '0' + month}`;
};

MgInputDate.args = {
  // Global
  value: `${date.getFullYear()}-${getMonth(date)}-${date.getDate()}`,
  identifier: 'identifier',
  name: 'input-name',
  // Label
  label: 'Label',
  labelOnTop: false,
  labelHide: false,
  // Input
  required: true,
  readonly: false,
  disabled: false,
  // Tooltip
  tooltip: 'This is a tooltip',
  // date range
  min: `${date.getFullYear()}-01-01`,
  max: `${date.getFullYear() + 1}-12-31`,
  // Help Text
  helpText: 'Help text with html <strong>bold</strong>, <em>italic</em>.',
};
