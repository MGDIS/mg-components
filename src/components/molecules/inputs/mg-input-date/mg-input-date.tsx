import { Component, Event, EventEmitter, h, Prop, State, Watch } from '@stencil/core';
import { MgInput } from '../MgInput';
import { createID, ClassList } from '../../../../utils/components.utils';
import { localeDate, dateRegExp } from '../../../../utils/locale.utils';
import { messages } from '../../../../locales';

@Component({
  tag: 'mg-input-date',
  styleUrl: './mg-input-date.scss',
  shadow: true,
})
export class MgInputDate {
  /************
   * Internal *
   ************/

  // classes
  private classError = 'is-not-valid';

  // HTML selector
  private input: HTMLInputElement;

  /**************
   * Decorators *
   **************/

  /**
   * Component value
   */
  @Prop({ mutable: true, reflect: true }) value: string;
  @Watch('value')
  validateValue(newValue) {
    if (newValue !== undefined && newValue !== '' && !(typeof newValue === 'string' && dateRegExp.test(newValue))) {
      throw new Error("<mg-input-date> props 'value' doesn't match pattern: yyyy-mm-dd");
    }
  }

  /**
   * Identifier is used for the element ID (id is a reserved prop in Stencil.js)
   * If not set, it will be created.
   */
  @Prop() identifier?: string = createID('mg-input-date');

  /**
   * Input name
   * If not set the value equals the identifier
   */
  @Prop() name?: string = this.identifier;

  /**
   * Input label
   * Required
   */
  @Prop() label!: string;

  /**
   * Define if label is displayed on top
   */
  @Prop() labelOnTop: boolean;

  /**
   * Define if label is visible
   */
  @Prop() labelHide: boolean = false;

  /**
   * Define if input is required
   */
  @Prop() required: boolean = false;

  /**
   * Define if input is readonly
   */
  @Prop() readonly: boolean = false;

  /**
   * Define if input is disabled
   */
  @Prop() disabled: boolean = false;

  /**
   * Add a tooltip message next to the input
   */
  @Prop() tooltip: string;

  /**
   * Add a help text under the input, usually expected data format and example
   */
  @Prop() helpText: string;

  /**
   * Define input pattern to validate
   */
  @Prop({ mutable: true }) valid: boolean;

  /**
   * Define input pattern error message
   */
  @Prop({ mutable: true }) invalid: boolean;

  /**
   * Define input minimum date
   * format: yyyy-mm-dd
   */
  @Prop() min: string;
  @Watch('min')
  validateMin(newValue) {
    this.validateDateFormat(newValue);
  }

  /**
   * Define input maximum date
   * format: yyyy-mm-dd
   */
  @Prop() max: string;
  @Watch('max')
  validateMax(newValue) {
    this.validateDateFormat(newValue);
  }

  /**
   * Component classes
   */
  @State() classList: ClassList = new ClassList(['mg-input--date']);

  /**
   * Error message to display
   */
  @State() errorMessage: string;

  /**
   * Emmited event when value change
   */
  @Event({ eventName: 'value-change' }) valueChange: EventEmitter<string>;

  /**
   * Handle input event
   */
  private handleInput = () => {
    this.checkValidity();
    this.value = this.input.value;
    this.valueChange.emit(this.value);
  };

  /**
   * Handle blur event
   */
  private handleBlur = () => {
    this.checkValidity();
    this.checkError();
  };

  /**
   * Date format validation
   * @param date
   */
  private validateDateFormat(date: string) {
    if (date?.length > 0 && !(typeof date === 'string' && dateRegExp.test(date))) {
      throw new Error("<mg-input-date> props 'min/max' doesn't match pattern: yyyy-mm-dd");
    }
  }

  /**
   * Check if input is valid
   */
  private checkValidity() {
    if (!this.readonly && this.input !== undefined) {
      const validity = this.input.checkValidity();

      // Set validity
      this.valid = validity;
      this.invalid = !validity;
    }
  }

  /**
   * Set error message
   */
  private setErrorMessage() {
    // required
    if (this.input.validity.valueMissing) {
      this.errorMessage = messages.errors.required;
    }
    // min & max
    else if ((this.input.validity.rangeUnderflow || this.input.validity.rangeOverflow) && this.min?.length > 0 && this.max?.length > 0) {
      this.errorMessage = messages.errors.date.minMax.replace('{min}', localeDate(this.min)).replace('{max}', localeDate(this.max));
    }
    // min
    else if (this.input.validity.rangeUnderflow) {
      this.errorMessage = messages.errors.date.min.replace('{min}', localeDate(this.min));
    }
    //max
    else if (this.input.validity.rangeOverflow) {
      this.errorMessage = messages.errors.date.max.replace('{max}', localeDate(this.max));
    }
    // wrong date format
    // element.validity.badInput is default error message
    else {
      this.errorMessage = messages.errors.date.badInput.replace('{min}', this.min?.length > 0 ? localeDate(this.min) : localeDate('1900-01-01'));
    }
  }

  /**
   * Check input errors
   */
  private checkError() {
    // Set error message
    this.errorMessage = undefined;
    if (!this.valid) {
      this.setErrorMessage();
      this.classList.add(this.classError);
    } else {
      this.classList.delete(this.classError);
    }
  }

  /*************
   * Lifecycle *
   *************/

  componentWillLoad() {
    this.validateValue(this.value);
    this.validateMin(this.min);
    this.validateMax(this.max);

    // return a promise tu process action only in the FIRST render().
    // https://stenciljs.com/docs/component-lifecycle#componentwillload
    return setTimeout(() => this.checkValidity.bind(this)(), 0);
  }

  render() {
    return (
      <MgInput
        identifier={this.identifier}
        classList={this.classList}
        label={this.label}
        labelOnTop={this.labelOnTop}
        labelHide={this.labelHide}
        required={this.required}
        readonly={this.readonly}
        value={this.value}
        readonlyValue={localeDate(this.value)}
        tooltip={this.tooltip}
        displayCharacterLeft={undefined}
        characterLeftTemplate={undefined}
        maxlength={undefined}
        helpText={this.helpText}
        errorMessage={this.errorMessage}
        isFieldset={false}
      >
        <input
          type="date"
          class="mg-input__box"
          min={this.min}
          max={this.max}
          value={this.value}
          id={this.identifier}
          name={this.name}
          disabled={this.disabled}
          required={this.required}
          onInput={this.handleInput}
          onBlur={this.handleBlur}
          pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}"
          ref={el => (this.input = el as HTMLInputElement)}
        />
      </MgInput>
    );
  }
}
