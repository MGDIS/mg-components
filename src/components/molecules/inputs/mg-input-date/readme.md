# mg-input-date

<!-- Auto Generated Below -->


## Properties

| Property             | Attribute      | Description                                                                                                 | Type      | Default                     |
| -------------------- | -------------- | ----------------------------------------------------------------------------------------------------------- | --------- | --------------------------- |
| `disabled`           | `disabled`     | Define if input is disabled                                                                                 | `boolean` | `false`                     |
| `helpText`           | `help-text`    | Template to use for characters left sentence                                                                | `string`  | `undefined`                 |
| `identifier`         | `identifier`   | Identifier is used for the element ID (id is a reserved prop in Stencil.js) If not set, it will be created. | `string`  | `createID('mg-input-date')` |
| `invalid`            | `invalid`      | Define input pattern error message                                                                          | `boolean` | `undefined`                 |
| `label` _(required)_ | `label`        | Input label Required                                                                                        | `string`  | `undefined`                 |
| `labelColon`         | `label-colon`  | Define if label has colon ":"                                                                               | `boolean` | `false`                     |
| `labelOnTop`         | `label-on-top` | Define if label is displayed on top                                                                         | `boolean` | `undefined`                 |
| `name`               | `name`         | Input name If not set the value equals the identifier                                                       | `string`  | `this.identifier`           |
| `readonly`           | `readonly`     | Define if input is readonly                                                                                 | `boolean` | `false`                     |
| `required`           | `required`     | Define if input is required                                                                                 | `boolean` | `false`                     |
| `tooltip`            | `tooltip`      | Add a tooltip message next to the input                                                                     | `string`  | `undefined`                 |
| `valid`              | `valid`        | Define input pattern to validate                                                                            | `boolean` | `undefined`                 |
| `value`              | `value`        | Component value                                                                                             | `string`  | `undefined`                 |


## Events

| Event         | Description                     | Type                  |
| ------------- | ------------------------------- | --------------------- |
| `valueChange` | Emmited event when value change | `CustomEvent<string>` |


## Dependencies

### Depends on

- [mg-label](../../../atoms/mg-label)
- [mg-tooltip](../../../atoms/mg-tooltip)
- [mg-icon](../../../atoms/mg-icon)
- [mg-character-left](../../../atoms/mg-character-left)
- [mg-help-text](../../../atoms/mg-help-text)

### Graph
```mermaid
graph TD;
  mg-input-date --> mg-label
  mg-input-date --> mg-tooltip
  mg-input-date --> mg-icon
  mg-input-date --> mg-character-left
  mg-input-date --> mg-help-text
  style mg-input-date fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
