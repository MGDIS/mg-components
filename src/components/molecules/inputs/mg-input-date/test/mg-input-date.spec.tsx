import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgInputDate } from '../mg-input-date';
import locale from '../../../../../locales';

const getPage = (args) => newSpecPage({
  components: [MgInputDate],
  template: () => (<mg-input-date {...args}></mg-input-date>)
});

describe('mg-input-date', () => {

  /**
   * Snapshots
   */
  test.each([
    {label: 'label', identifier: "identifier"},
    {label: 'label', identifier: "identifier", labelOnTop: true},
  ])('Should render with args %s:', async (args) => {
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });

  /**
   * Test
   */

   test.each(["", undefined])('Should throw an error with invalid label property : %s', async (value) => {
    try {
      await getPage({label:value});
    }
    catch (err) {
      expect(err.message).toMatch('prop "label" is required')
    }
  });

  test.each(["", undefined])('Should throw an error when pattern is used with patternErrorMessage: %s', async (value) => {
    try {
      const { root } = await getPage({label: "blu", pattern:'[a-z]*', patternErrorMessage: value});
      expect(root).toMatchSnapshot();
    }
    catch (err) {
      expect(err.message).toMatch('<mg-input-text> prop "pattern" must be paired with the prop "patternErrorMessage"')
    }
  });

   test('Should trigger events', async ()=> {
    const inputValue = '2021-10-14';
    const args = {label: 'label', helpText: "My Help Text", identifier: "identifier", required: true};
    const page = await getPage(args);

    const element = await page.doc.querySelector('mg-input-date');
    const input = element.shadowRoot.querySelector('input');

    jest.spyOn(page.rootInstance.inputChange, 'emit');

    input.dispatchEvent(new CustomEvent('focus', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.classes.has('is-focused')).toBeTruthy();

    expect(page.root).toMatchSnapshot(); //Snapshot on focus

    input.value = inputValue;
    input.dispatchEvent(new CustomEvent('input', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.inputChange.emit).toHaveBeenCalledWith(inputValue);

    input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.classes.has('is-focused')).toBeFalsy();
  });

  test.each([
    {validity: true, valueMissing: false, badInput: false},
    {validity: false, valueMissing: true, badInput: false},
    {validity: false, valueMissing: false, badInput: true},
  ])('validity (%s), valueMissing (%s), badInput (%s)', async ({validity, valueMissing, badInput})=> {
    const args = {label: 'label', helpText: "My Help Text", identifier: "identifier"};
    const page = await getPage(args);

    const element = await page.doc.querySelector('mg-input-date');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(()=> validity);
    Object.defineProperty(input, 'validity', { get: jest.fn(()=> ({
      valueMissing,
      badInput
    }))});

    input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
    await page.waitForChanges();

    if(validity) {
      expect(page.rootInstance.errorMessage).toBeUndefined();
    }
    else if (valueMissing){
      expect(page.rootInstance.errorMessage).toEqual(locale.errors.required);
    }
    else if(badInput) {
      expect(page.rootInstance.errorMessage).toEqual(locale.errors.date.badInput);
    }
    expect(page.rootInstance.valid).toEqual(validity);
    expect(page.rootInstance.invalid).toEqual(!validity);
  });
});
