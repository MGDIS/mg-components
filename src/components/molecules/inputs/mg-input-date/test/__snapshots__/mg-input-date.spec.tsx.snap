// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-input-date Should render with args {
  label: 'label',
  identifier: 'identifier',
  readonly: true,
  labelOnTop: true,
  tooltip: 'Tooltip message'
}: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date mg-input--label-on-top mg-input--readonly">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
      </div>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args {
  label: 'label',
  identifier: 'identifier',
  readonly: true,
  value: '2021-10-15'
}: 1`] = `
<mg-input-date value="2021-10-15">
  <mock:shadow-root>
    <div class="mg-input mg-input--date mg-input--readonly">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong>
          15/10/2021
        </strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args {
  label: 'label',
  identifier: 'identifier',
  tooltip: 'My Tooltip Message'
}: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
          <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
            <mg-icon icon="info-circle"></mg-icon>
          </mg-tooltip>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args {
  label: 'label',
  identifier: 'identifier',
  tooltip: 'My Tooltip Message',
  labelOnTop: true
}: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date mg-input--label-on-top">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
        <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
          <mg-icon icon="info-circle"></mg-icon>
        </mg-tooltip>
      </div>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args { label: 'label', identifier: 'identifier' }: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args { label: 'label', identifier: 'identifier', labelHide: true }: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date">
      <mg-input-title class="sr-only" identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args { label: 'label', identifier: 'identifier', labelOnTop: true }: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date mg-input--label-on-top">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
      </div>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should render with args { label: 'label', identifier: 'identifier', readonly: true }: 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date mg-input--readonly">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date Should trigger events 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-help-text" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
        <div class="mg-input__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date validity, case next state is disabled validity ({ validity: false, valueMissing: true, badInput: false }), valueMissing (%s), badInput (%s) 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--date">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          Ce champ est obligatoire.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date validity, case next state is disabled validity ({ validity: false, valueMissing: true, badInput: false }), valueMissing (%s), badInput (%s) 2`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" disabled="" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          Ce champ est obligatoire.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date validity, case next state is readonly validity ({ validity: false, valueMissing: true, badInput: false }), valueMissing (%s), badInput (%s) 1`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="is-not-valid mg-input mg-input--date">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <input aria-describedby="identifier-error" class="mg-input__box" id="identifier" name="identifier" pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}" type="date">
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          Ce champ est obligatoire.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;

exports[`mg-input-date validity, case next state is readonly validity ({ validity: false, valueMissing: true, badInput: false }), valueMissing (%s), badInput (%s) 2`] = `
<mg-input-date>
  <mock:shadow-root>
    <div class="mg-input mg-input--date mg-input--readonly">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-date>
`;
